body.tex

```diff
@@ -268,14 +268,14 @@ \subsection{The Fund}
 \section{Projects}
 A project $P$ is a tuple:
 \[
-    P = \tuple{\field{P}{name}, \field{P}{org}, \field{P}{k}, \field{P}{meta}}
+    P = \tuple{\field{P}{id}, \field{P}{k}, \field{P}{meta}}
 \]
 
 \spec{
     \definition
     \begin{itemize}
-        \item $\field{P}{name}$ is the unique project name within the context of an $\field{P}{org}$,
```

Comment:
So, currently, we'd have something like "radicle.monadic" where "radicle" is the project name and "monadic" the org id.
Does this mean that, given this new spec, we can have "radicle.monadic" and/or "radicle.cloudhead", where "cloudhead" would be a user?

Reply:
Exactly!

Reply:
This question is answered if you look at the definition of a project id. Which is comprised of an `either` of user or org. Therefore you can have `(Org("monadic"), "radicle")` and another `(User("nuno"), "radicle")`.

Reply:
right. Speaking of code, now that the project id is, as you say, an either, having more type safety is essential. Now a project has a name and an org_id, both being `String32`, which already gives me the chills. The new id  type should capture and keep us from evil:

``` haskell
data ProjectId 
  = UserProject UserId ProjectName 
  | OrgProject OrgId ProjectName 
```

or

``` haskell
data ProjectOwner = Org OrgId | User UserId
data ProjectId = ProjectId ProjectOwner ProjectName

Reply:
This looks like the right direction, but we should have those discussions in the context of the implementation down the line.

Reply:
Absolutely 🙌

Reply:
Yeah I would definitely have an opaque `OrgId` and `UserId` type to prevent mis-use. I would also consider mapping unique ids with numbers/indices like you would do in a traditional relational setting. Eg. have a u64 counter that is incremented everytime an org is registered, and use that as the `OrgId`. I think substrate does this for account ids. But let's have this discussion under `radicle-registry` :+1: